staticaddr/deposit/manager.go

```diff
@@ -390,115 +375,152 @@ func (m *Manager) filterNewDeposits(utxos []*lnwallet.Utxo) []*lnwallet.Utxo {
 	// Create a state machine for a given deposit.
-	fsm, err := NewFSM(
-		m.runCtx, deposit, m.cfg, m.finalizedDepositChan, false,
-	)
+	fsm, err := NewFSM(ctx, deposit, m.cfg, m.finalizedDepositChan, false)
 	if err != nil {
 		return err
 	}
 
 	// Send the start event to the state machine.
 	go func() {
-		err = fsm.SendEvent(m.runCtx, OnStart, nil)
+		err = fsm.SendEvent(ctx, OnStart, nil)
 		if err != nil {
 			log.Errorf("Error sending OnStart event: %v", err)
 		}
 	}()
 
-	err = fsm.DefaultObserver.WaitForState(m.runCtx, time.Minute, Deposited)
+	err = fsm.DefaultObserver.WaitForState(ctx, time.Minute, Deposited)
 	if err != nil {
 		return err
 	}
 
 	// Add the FSM to the active FSMs map.
-	m.Lock()
+	m.mu.Lock()
 	m.activeDeposits[deposit.OutPoint] = fsm
-	m.Unlock()
+	m.mu.Unlock()
 
 	return nil
 }
 
 func (m *Manager) finalizeDeposit(outpoint wire.OutPoint) {
-	m.Lock()
+	m.mu.Lock()
 	delete(m.activeDeposits, outpoint)
-	delete(m.deposits, outpoint)
-	m.Unlock()
+	m.mu.Unlock()
 }
 
-// GetActiveDepositsInState returns all active deposits.
+// GetActiveDepositsInState returns all active deposits. This function is called
+// on a client restart before the manager is fully initialized, hence we don't
+// have to lock the deposits.
 func (m *Manager) GetActiveDepositsInState(stateFilter fsm.StateType) (
 	[]*Deposit, error) {
 
-	m.Lock()
-	defer m.Unlock()
+	m.mu.Lock()
+	defer m.mu.Unlock()
 
 	var deposits []*Deposit
 	for _, fsm := range m.activeDeposits {
-		if fsm.deposit.GetState() != stateFilter {
+		deposits = append(deposits, fsm.deposit)
+	}
+
+	lockDeposits(deposits)
+	defer unlockDeposits(deposits)
+
+	filteredDeposits := make([]*Deposit, 0, len(deposits))
+	for _, d := range deposits {
+		if !d.IsInStateNoLock(stateFilter) {
 			continue
 		}
-		deposits = append(deposits, fsm.deposit)
+
+		filteredDeposits = append(filteredDeposits, d)
 	}
 
-	sort.Slice(deposits, func(i, j int) bool {
-		return deposits[i].ConfirmationHeight <
-			deposits[j].ConfirmationHeight
+	sort.Slice(filteredDeposits, func(i, j int) bool {
+		return filteredDeposits[i].ConfirmationHeight <
+			filteredDeposits[j].ConfirmationHeight
 	})
 
-	return deposits, nil
-}
-
-// GetAllDeposits returns all active deposits.
-func (m *Manager) GetAllDeposits() ([]*Deposit, error) {
-	return m.cfg.Store.AllDeposits(m.runCtx)
+	return filteredDeposits, nil
 }
 
 // AllOutpointsActiveDeposits checks if all deposits referenced by the outpoints
-// are active and in the specified state.
+// are in out in-mem active deposits map and in the specified state. If
+// fsm.EmptyState is set as targetState all deposits are returned regardless of
+// their state. Each existent deposit is locked during the check.
 func (m *Manager) AllOutpointsActiveDeposits(outpoints []wire.OutPoint,
-	stateFilter fsm.StateType) ([]*Deposit, bool) {
+	targetState fsm.StateType) ([]*Deposit, bool) {
 
-	m.Lock()
-	defer m.Unlock()
+	_, deposits := m.toActiveDeposits(&outpoints)
```

Comment:
You are right, there's a gap in coverage. It could be that a deposit is removed from the active map after the `toActiveDeposits` call.
Instead, we can `m.mu.Lock()` throughout the `AllOutpointsActiveDeposits` call and in other calls sites.